### PR TITLE
Download manifest and favicon.ico always

### LINKED ISSUE #71: All collectors should automatically download the same type of resources (scripts, styles, fonts, icons, etc.)
By default, the collectors will not request certain resources (e.g.: manifest file, all font files, etc.). 

Since quite a few rules will need to analyze those requests/resources, it wouldn't make sense to have every rule add custom code to request them, so we should add a helper to do that and notify the subscribed rules.

---

What we might want to request (incomplete list):

- <del>manifest file (`<link rel="manifest" href="site.webmanifest">`)</del>
- `rss` and `atom` files (`<link rel="alternate" ... href="...">`)
- font files (specified in `@font-face` rules)
- images specified in css files, manifest file, etc.
- video, audio, and subtitle files
- source map files (e.g.: `//# sourceMappingURL=example.js.map`)

### PR DESCRIPTION
**Pull request by sarvaje**
Fix #71

### CODE REVIEW

File: src/lib/connectors/jsdom/jsdom.ts [hunk 1]
```diff
@@ -215,6 +220,22 @@ class JSDOMConnector implements IConnector {
...
 
+    /** JSDOM doesn't download the favicon automatically, this method:
+     *
+     * * uses the `src` attribute of `<link rel="icon">` if present.
+     * * uses `favicon.ico` and the final url after redirects.
+     */
+    private async getFavicon(element?: HTMLElement) {
+        const href = element ? element.getAttribute('href') : '/favicon.ico';
+
+        //TODO: remove "as any" once https://github.com/DefinitelyTyped/DefinitelyTyped/issues/16860 is fixed
```
Comment (molant) [inline src/lib/connectors/jsdom/jsdom.ts]: With the latest package update I think this is no longer needed. Can you test it?

File: src/lib/connectors/shared/remote-debugging-connector.ts [hunk 2]
```diff
@@ -484,6 +481,42 @@ export class Connector implements IConnector {
...
+                error,
+                hops,
+                resource: href
+            };
+
+            await this._server.emitAsync('fetch::error', event);
+        }
+    }
+
     /** Initiates all the proce */
```
Comment (molant) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: Typo in the comment from before. Can you fix it?

**Comment by sarvaje** [comment]
@molant done!

File: src/lib/connectors/shared/remote-debugging-connector.ts [hunk 3]
```diff
@@ -484,7 +481,43 @@ export class Connector implements IConnector {
         ]);
     }
 
-    /** Initiates all the proce */
+    /** CDP sometimes doesn't download the favicon automatically, this method:
+ *
```
Comment (alrra) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: The comment is a bit off.

File: src/lib/connectors/shared/remote-debugging-connector.ts [hunk 4]
```diff
@@ -332,12 +325,16 @@ export class Connector implements IConnector {
             };
         }
 
+        if (data.element && data.element.nodeName.toLowerCase() === 'link' && data.element.getAttribute('rel') === 'icon') {
```
Comment (alrra) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: The `data.element.getAttribute('rel') === 'icon'` check will fail for something like [`REL="SHORTCUT ICON"`](https://mathiasbynens.be/notes/rel-shortcut-icon).

File: src/lib/connectors/shared/remote-debugging-connector.ts (hunk 3 above)
Comment (sarvaje) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: @alrra what do you mean with "a bit off"?

Comment (alrra) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: @sarvaje The indentation is not correct.

File: src/lib/connectors/shared/remote-debugging-connector.ts (hunk 4 above)
Comment (sarvaje) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: This is done, take a look to the new implementation :)

File: src/lib/connectors/shared/remote-debugging-connector.ts (hunk 3 above)
Comment (sarvaje) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: ohhh I see hahaha

Comment (sarvaje) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: done!

**Comment by sarvaje** [comment]
@alrra all done!

File: src/lib/connectors/shared/remote-debugging-connector.ts [hunk 5]
```diff
@@ -291,6 +284,25 @@ export class Connector implements IConnector {
         return response;
     }
 
+    private isFaviconLink(element: IAsyncHTMLElement) {
```
Comment (alrra) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: @sarvaje Maybe make this a more general thing (e.g.: `hasAttributeWithValue(element, value)`) , and move it into `utils/misc`?

Comment (sarvaje) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: ok!

Comment (sarvaje) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: done!

**Pull request closed without merging**